### ISSUE TITLE
Custom keywords

### ISSUE THREAD

**Issue by atomictag** [leadpony/justify#16]
Hi there, I know justify does not support custom keywords (which btw would be an awesome feature to add alongside the current _custom formats_). However, it could be useful to treat custom keywords differently than today - i.e. if a keyword is unrecognised, leave the value untouched instead of reading its schema and evaluate it, which most of times makes the whole keyword useless. This way it would still be possible for user code to do something useful with that - e.g. lookup the keyword on demand from a schema.

For example:
```
{ "type" : "string", "errorMessage" : "some message here"}
```
produces the JsonSchema
```
{ "type" : "string" }
```
Similarly:
```
{ "type" : "string", "errorMessage" : { "type" : "some message here"} }
```
produces the JsonSchema
```
{ "type" : "string", "errorMessage" : {}}
```
If the above values were left untouched custom code could, for example, lookup the specific error message from a failing schema reported by _Problem.getSchema()_ and use that value.

The goal I am trying to achieve is to do something similar to what _Ajv_ does with https://github.com/epoberezkin/ajv-errors - without extending justify to support that.
This would be very useful in my use case to provide domain-specific messages directly in the schema rather than globally.

**Comment by leadpony** [comment]
Thank you @atomictag again. I am interesting in _domain-specific messages_ because validators often emit error messeages not best fit for most of end-users.
My original intention was to leave unrecognized keywords intact in the schema read, but current implementation may not work for the purpose. I will check it later.

**Comment by leadpony** [comment]
I have reproduced the problem. Thank you for your reporting.
The current implementation ignores unrecognized keywords that can not be reused as JSON schemas.
I will make the implementation better.

**Comment by maheshcamath** [comment]
Any workaround for this? I need this library to validate and chase the references but its ignoring all my custom properties.

**Comment by leadpony** [comment]
I know this is not a good workaround, but the following schema may help you.
```json
{ "yourCustomKeyword": {"pattern": "yourMessageHere"} }
```
The value shown above can be obtained using `JsonSchema.toJson()`.
This issue will be resolved in the next release which will be available in 2 weeks.
Sorry for the inconvenience.

**Comment by atomictag** [comment]
Thank you for your efforts @leadpony ! Can't wait to try out the new 0.17.0 release.

**Comment by leadpony** [comment]
Hello @atomictag 
Yes I have released 0.17.0, but this release resolves only the half of the reported issue. Even with the release, any keyword reserved by the JSON Schema specification, which has an invalid value according to the specification, is not retained in the loaded schema. This limitation will be fixed in the future release.
For other important notice, please refer to the [CHANGELOG](https://github.com/leadpony/justify/blob/master/CHANGELOG.md).
I believe the support of real custom keywords will be a long-term goal of this library.
Thank you so much for your useful suggestion.

**Comment by leadpony** [comment]
The reported bug is now completely fixed in the release 1.0.0. Custom keyword feature should be supported in the future release.
